src/checkbox.jsx

```diff
@@ -55,7 +55,10 @@ const Checkbox = React.createClass({
   //from the parent / owner using context
   componentWillReceiveProps(nextProps, nextContext) {
     let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
-    this.setState({muiTheme: newMuiTheme});
+    this.setState({
+      muiTheme: newMuiTheme,
+      switched: nextProps.hasOwnProperty('checked') && nextProps.checked !== this.state.switched,
```

Comment:
Following this reverted PR https://github.com/callemall/material-ui/pull/2303.
What do you think about using instead
```js
switched: this.props.checked !== nextProps.checked,
```